apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx

```diff
@@ -0,0 +1,183 @@
+import AccountAddress from '_components/account-address';
+import Alert from '_components/alert';
+import Loading from '_components/loading';
+import RecentTransactions from '_components/transactions-card/RecentTransactions';
+import { SuiIcons } from '_font-icons/output/sui-icons';
+import { useAppSelector, useObjectsState } from '_hooks';
+import { accountAggregateBalancesSelector } from '_redux/slices/account';
+import { GAS_TYPE_ARG, Coin } from '_redux/slices/sui-objects/Coin';
+
+import st from './TokensPage.module.scss';
+
+type TokenDetailsProps = {
+    coinType?: string;
+};
+
+const emptyWalletDescription = (
+    <div className={st.emptyWalletDescription}>
+        To conduct transactions on the Sui network, you need SUI in your wallet.
+    </div>
+);
+
+type TokensProps = {
+    allCoinTypes: string[];
+    coinBalance: bigint;
+    balances: Record<string, bigint>;
+    loading: boolean;
+};
+
+function MyTokens({
+    allCoinTypes,
+    coinBalance,
+    balances,
+    loading,
+}: TokensProps) {
+    return (
+        <Loading loading={loading} className={st.othersLoader}>
+            {allCoinTypes.length ? (
+                <>
+                    <div className={st.title}>MY COINS</div>
+                    <div className={st.otherCoins}>
+                        {allCoinTypes.map((aCoinType) => (
+                            <CoinBalance
+                                type={aCoinType}
+                                balance={balances[aCoinType] || BigInt(0)}
+                                key={aCoinType}
+                            />
+                        ))}
+                        {coinBalance <= 0 ? (
+                            <div className={st.emptyWallet}>
+                                <FaucetRequestButton />
+                                {emptyWalletDescription}
+                            </div>
+                        ) : null}
+                    </div>
+                </>
+            ) : (
+                <div className={st.emptyWallet}>
+                    <div className={st.emptyWalletIcon} />
+                    <div className={st.emptyWalletTitle}>
+                        Your wallet contains no SUI.
+                    </div>
+                    {emptyWalletDescription}
+                    <FaucetRequestButton />
+                </div>
+            )}
+            <FaucetMessageInfo className={st.gasRequestInfo} />
+        </Loading>
+    );
+}
+
+function TokenDetails({ coinType }: TokenDetailsProps) {
+    const { loading, error, showError } = useObjectsState();
+    const activeCoinType = coinType || GAS_TYPE_ARG;
+    const balances = useAppSelector(accountAggregateBalancesSelector);
+    const tokenBalance = balances[activeCoinType] || BigInt(0);
+    const allCoinTypes = useMemo(() => Object.keys(balances), [balances]);
+    const coinTypeWithBalance =
+        !coinType && tokenBalance > 0 ? activeCoinType : allCoinTypes[0];
```

Comment:
```suggestion
        coinType || tokenBalance > 0 ? activeCoinType : allCoinTypes[0];
```

Otherwise I think when we provide a coinType there is a chance to select a different coin